### ISSUE TITLE
Code coverage no longer works

### ISSUE THREAD

**Issue by maxbennedich** [nep-pack/NonlinearEigenproblems.jl#115]
Seems to have stopped working around September 20. From the build log:
```
Submitting data to Coveralls...
┌ Warning: Error in @async writebody task
│   exception =
│    MethodError: no method matching write(::HTTP.Streams.Stream{HTTP.Messages.Response,HTTP.ConnectionPool.Transaction{MbedTLS.SSLContext}}, ::Pair{String,HTTP.Multipart{Base.GenericIOBuffer{Array{UInt8,1}}}})
│    Closest candidates are:
│      write(::IO, ::Any) at io.jl:498
│      write(::IO, ::Any, !Matched::Any...) at io.jl:500
│      write(::IO, !Matched::Complex) at complex.jl:217
│      ...
│    Stacktrace:
│     [1] write(::HTTP.Streams.Stream{HTTP.Messages.Response,HTTP.ConnectionPool.Transaction{MbedTLS.SSLContext}}, ::Pair{String,HTTP.Multipart{Base.GenericIOBuffer{Array{UInt8,1}}}}) at ./io.jl:498
│     [2] writechunk(::HTTP.Streams.Stream{HTTP.Messages.Response,HTTP.ConnectionPool.Transaction{MbedTLS.SSLContext}}, ::HTTP.Messages.Request, ::Pair{String,HTTP.Multipart{Base.GenericIOBuffer{Array{UInt8,1}}}}) at /home/travis/.julia/packages/HTTP/mwR9J/src/StreamRequest.jl:129
│     [3] writebodystream at /home/travis/.julia/packages/HTTP/mwR9J/src/StreamRequest.jl:119 [inlined]
│     [4] writebody(::HTTP.Streams.Stream{HTTP.Messages.Response,HTTP.ConnectionPool.Transaction{MbedTLS.SSLContext}}, ::HTTP.Messages.Request, ::Dict{String,HTTP.Multipart{Base.GenericIOBuffer{Array{UInt8,1}}}}) at /home/travis/.julia/packages/HTTP/mwR9J/src/StreamRequest.jl:97
│     [5] (::getfield(HTTP.StreamRequest, Symbol("##2#3")){HTTP.ConnectionPool.Transaction{MbedTLS.SSLContext},HTTP.Messages.Request,Dict{String,HTTP.Multipart{Base.GenericIOBuffer{Array{UInt8,1}}}}})() at ./task.jl:259
└ @ HTTP.StreamRequest ~/.julia/packages/HTTP/mwR9J/src/StreamRequest.jl:60
ERROR: LoadError: MethodError: no method matching write(::HTTP.Streams.Stream{HTTP.Messages.Response,HTTP.ConnectionPool.Transaction{MbedTLS.SSLContext}}, ::Pair{String,HTTP.Multipart{Base.GenericIOBuffer{Array{UInt8,1}}}})
Closest candidates are:
  write(::IO, ::Any) at io.jl:498
  write(::IO, ::Any, !Matched::Any...) at io.jl:500
  write(::IO, !Matched::Complex) at complex.jl:217
  ...
Stacktrace:
 [1] #request#1(::Nothing, ::Nothing, ::Int64, ::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::Type{HTTP.StreamRequest.StreamLayer}, ::HTTP.ConnectionPool.Transaction{MbedTLS.SSLContext}, ::HTTP.Messages.Request, ::Dict{String,HTTP.Multipart{Base.GenericIOBuffer{Array{UInt8,1}}}}) at ./task.jl:244
 [2] (::getfield(HTTP, Symbol("#kw##request")))(::NamedTuple{(:iofunction,),Tuple{Nothing}}, ::typeof(HTTP.request), ::Type{HTTP.StreamRequest.StreamLayer}, ::HTTP.ConnectionPool.Transaction{MbedTLS.SSLContext}, ::HTTP.Messages.Request, ::Dict{String,HTTP.Multipart{Base.GenericIOBuffer{Array{UInt8,1}}}}) at ./none:0
 [3] #request#1(::Nothing, ::Type, ::Int64, ::Base.Iterators.Pairs{Symbol,Nothing,Tuple{Symbol},NamedTuple{(:iofunction,),Tuple{Nothing}}}, ::Function, ::Type{HTTP.ConnectionRequest.ConnectionPoolLayer{HTTP.StreamRequest.StreamLayer}}, ::HTTP.URIs.URI, ::HTTP.Messages.Request, ::Dict{String,HTTP.Multipart{Base.GenericIOBuffer{Array{UInt8,1}}}}) at /home/travis/.julia/packages/HTTP/mwR9J/src/ConnectionRequest.jl:52
 [4] #request at ./none:0 [inlined]
 [5] #request#1 at /home/travis/.julia/packages/HTTP/mwR9J/src/ExceptionRequest.jl:19 [inlined]
 [6] (::getfield(HTTP, Symbol("#kw##request")))(::NamedTuple{(:iofunction,),Tuple{Nothing}}, ::typeof(HTTP.request), ::Type{HTTP.ExceptionRequest.ExceptionLayer{HTTP.ConnectionRequest.ConnectionPoolLayer{HTTP.StreamRequest.StreamLayer}}}, ::HTTP.URIs.URI, ::HTTP.Messages.Request, ::Dict{String,HTTP.Multipart{Base.GenericIOBuffer{Array{UInt8,1}}}}) at ./none:0
 [7] (::getfield(Base, Symbol("###44#45#46")){ExponentialBackOff,getfield(HTTP.RetryRequest, Symbol("##2#3")){Bool,HTTP.Messages.Request},typeof(HTTP.request)})(::Base.Iterators.Pairs{Symbol,Nothing,Tuple{Symbol},NamedTuple{(:iofunction,),Tuple{Nothing}}}, ::Function, ::Type, ::Vararg{Any,N} where N) at ./error.jl:203
 [8] ##44#47 at ./none:0 [inlined]
 [9] #request#1(::Int64, ::Bool, ::Base.Iterators.Pairs{Symbol,Nothing,Tuple{Symbol},NamedTuple{(:iofunction,),Tuple{Nothing}}}, ::Function, ::Type{HTTP.RetryRequest.RetryLayer{HTTP.ExceptionRequest.ExceptionLayer{HTTP.ConnectionRequest.ConnectionPoolLayer{HTTP.StreamRequest.StreamLayer}}}}, ::HTTP.URIs.URI, ::HTTP.Messages.Request, ::Dict{String,HTTP.Multipart{Base.GenericIOBuffer{Array{UInt8,1}}}}) at /home/travis/.julia/packages/HTTP/mwR9J/src/RetryRequest.jl:44
 [10] #request at ./none:0 [inlined]
 [11] #request#1(::VersionNumber, ::String, ::Nothing, ::Nothing, ::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::Type{HTTP.MessageRequest.MessageLayer{HTTP.RetryRequest.RetryLayer{HTTP.ExceptionRequest.ExceptionLayer{HTTP.ConnectionRequest.ConnectionPoolLayer{HTTP.StreamRequest.StreamLayer}}}}}, ::String, ::HTTP.URIs.URI, ::Array{Pair{SubString{String},SubString{String}},1}, ::Dict{String,HTTP.Multipart{Base.GenericIOBuffer{Array{UInt8,1}}}}) at /home/travis/.julia/packages/HTTP/mwR9J/src/MessageRequest.jl:47
 [12] request at /home/travis/.julia/packages/HTTP/mwR9J/src/MessageRequest.jl:28 [inlined]
 [13] #request#1(::Int64, ::Bool, ::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::Type{HTTP.RedirectRequest.RedirectLayer{HTTP.MessageRequest.MessageLayer{HTTP.RetryRequest.RetryLayer{HTTP.ExceptionRequest.ExceptionLayer{HTTP.ConnectionRequest.ConnectionPoolLayer{HTTP.StreamRequest.StreamLayer}}}}}}, ::String, ::HTTP.URIs.URI, ::Array{Pair{SubString{String},SubString{String}},1}, ::Dict{String,HTTP.Multipart{Base.GenericIOBuffer{Array{UInt8,1}}}}) at /home/travis/.julia/packages/HTTP/mwR9J/src/RedirectRequest.jl:24
 [14] request(::Type{HTTP.RedirectRequest.RedirectLayer{HTTP.MessageRequest.MessageLayer{HTTP.RetryRequest.RetryLayer{HTTP.ExceptionRequest.ExceptionLayer{HTTP.ConnectionRequest.ConnectionPoolLayer{HTTP.StreamRequest.StreamLayer}}}}}}, ::String, ::HTTP.URIs.URI, ::Array{Pair{SubString{String},SubString{String}},1}, ::Dict{String,HTTP.Multipart{Base.GenericIOBuffer{Array{UInt8,1}}}}) at /home/travis/.julia/packages/HTTP/mwR9J/src/RedirectRequest.jl:21
 [15] #request#3(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::String, ::HTTP.URIs.URI, ::Array{Pair{SubString{String},SubString{String}},1}, ::Dict{String,HTTP.Multipart{Base.GenericIOBuffer{Array{UInt8,1}}}}) at /home/travis/.julia/packages/HTTP/mwR9J/src/HTTP.jl:302
 [16] request at /home/travis/.julia/packages/HTTP/mwR9J/src/HTTP.jl:302 [inlined]
 [17] #request#4 at /home/travis/.julia/packages/HTTP/mwR9J/src/HTTP.jl:316 [inlined]
 [18] #request at ./none:0 [inlined]
 [19] #post#8 at /home/travis/.julia/packages/HTTP/mwR9J/src/HTTP.jl:363 [inlined]
 [20] #post at ./none:0 [inlined] (repeats 2 times)
 [21] #submit#1(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::Array{FileCoverage,1}) at /home/travis/.julia/packages/Coverage/25tif/src/coveralls.jl:91
 [22] submit(::Array{FileCoverage,1}) at /home/travis/.julia/packages/Coverage/25tif/src/coveralls.jl:60
 [23] top-level scope at none:0
 [24] include at ./boot.jl:317 [inlined]
 [25] include_relative(::Module, ::String) at ./loading.jl:1038
 [26] include(::Module, ::String) at ./sysimg.jl:29
 [27] include(::String) at ./client.jl:388
 [28] top-level scope at none:0
in expression starting at /home/travis/build/nep-pack/NonlinearEigenproblems.jl/test/submit_test_coverage.jl:24
```

**Comment by jarlebring** [comment]
Isn't code coverage broken anyway since we switched to julia 1.0?

**Comment by maxbennedich** [comment]
Yes, the statistics is not correct since it misses some lines (github issue 28192 in JuliaLang/julia). The above seems to be a separate problem, where the submission of stats has suddenly started to crash.

**Comment by maxbennedich** [comment]
This seems to be a global issue, I've checked several other projects using Coveralls, and they all broke on September 20. I've opened an issue with Coverage.jl. Meanwhile, if we comment out the call to Coveralls that crashes, the Codecov submission still works, so I'll do that for now.

**Comment by maxbennedich** [comment]
Merged in #117 .
